internal/network/tcp_conn.go

```diff
@@ -0,0 +1,107 @@
+package network
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+
+	"golang.org/x/net/context"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	frameSizeBytes int = 4
+)
+
+var (
+	byteOrder = binary.BigEndian
+)
+
+var _ Conn = (*tcpConn)(nil)
+
+type tcpConn struct {
+	id         ID
+	underlying net.Conn
```

Comment:
Why `underlying`? What does it signify?

Reply:
Is it just because, `tcpConn.conn` will be a bit weird?

Reply:
It signifies that `tcpConn` wraps a `net.Conn` which is more clear when the wrapped component is called `underlying` or `wrapped` IMO